scripts: tests: Blackbox test expansion - coverage

KamilxPaszkiet opened the pull request:
Adds tests related to coverage flags:
`--coverage-tool`
`--coverage-basedir`
`--coverage-formats`

Move to coverage.py:
`--coverage`
`--enable-coverage`

LukaszMrugala commented on scripts/tests/twister_blackbox/test_coverage.py:
```diff
@@ -0,0 +1,294 @@
+#!/usr/bin/env python3
+# Copyright (c) 2023 Intel Corporation
```
2024, not 2023

LukaszMrugala commented on scripts/tests/twister_blackbox/test_coverage.py:
```diff
@@ -0,0 +1,294 @@
+
+import importlib
+import re
+
```
Remove blank line

LukaszMrugala commented on scripts/tests/twister_blackbox/test_coverage.py:
```diff
@@ -0,0 +1,294 @@
+            path = os.path.join(out_path, f_name)
+            assert os.path.exists(path), f'file not found {f_name}'
+
+        assert str(sys_exit.value) == '0'
```
Best to check the exit value in the first assert (if that breaks, you know more about the error).

LukaszMrugala commented on scripts/tests/twister_blackbox/test_coverage.py:
```diff
@@ -0,0 +1,294 @@
+        ]
+    )
+    def test_coverage(self, capfd, test_path, test_platforms, out_path, file_name):
+        args = ['-i','--outdir', out_path, '-T', test_path, '--coverage', '--coverage-tool', 'gcovr'] + \
```
I like to split args into separate lists, to have a clear picture of what are the flags under tests.

KamilxPaszkiet replied:
done, now it's more clear

fabiobaltieri commented:
@KamilxPaszkiet this needs a rebase

KamilxPaszkiet committed:
scripts: tests: Blackbox test expansion - coverage

Adds tests related to coverage flags:
--coverage-tool
--coverage-basedir
--coverage-formats

Move to coverage.py:
--coverage
--enable-coverage

Signed-off-by: Kamil Paszkiet <kamilx.paszkiet@intel.com>
